Sources/LiveViewNativeCore/Support.swift

```diff
@@ -97,23 +99,22 @@ public struct RustStrIterator: IteratorProtocol {
 }
 
 public protocol ToRustStr {
-    func toRustStr<T> (_ withUnsafeRustStr: (RustStr) -> T) -> T;
+    mutating func toRustStr<T> (_ withUnsafeRustStr: (RustStr) throws -> T) rethrows -> T
```

Comment:
`RustStr` is read-only, as it maps to Rust's `&str`; so I would expect that the public-facing API wouldn't carry any mutating functions, but I'm not super famiiar with the semantics of `mutating` in Swift, could you clarify whether this would violate that read-only property?

Reply:
For value types, functions need to be marked `mutating` in order to modify `self`. Only the implementation on `Swift.String` strictly needs to be `mutating`, but that means the protocol needs to allow `mutating` implementations. The implementation on `RustStr` still doesn't modify anything. I agree, it looks a bit weird API-wise, but I think it's fine.

That said, `RustStr` is a class, so even if the function isn't `mutating` (indeed, it can't be marked as such), it could still go ham.

Reply:
I read the commit message associated with this, and I think I understand now, this doesn't imply that the RustStr is being mutated, but rather that the implementor of the protocol may need to mutate itself in order to implement the function, correct? In that case, I think we're good.

Reply:
Exactly